src/tiff.cc

```diff
@@ -8,23 +8,22 @@
 
 namespace GdTiff {
 
-void tiff2img( std::vector< char > & data )
+void tiff2img( std::vector< char > & data, const char * format )
 {
   QImage img = QImage::fromData( (unsigned char *)&data.front(), data.size() );
 
   if ( !img.isNull() ) {
     QByteArray ba;
-
-    QBuffer buffer( &ba ); // buffer doesn't own ba
+    QBuffer buffer( &ba );
     buffer.open( QIODevice::WriteOnly );
-
     QSize screenSize = QApplication::primaryScreen()->availableSize();
     QSize imgSize    = img.size();
     int scaleSize    = qMin( imgSize.width(), screenSize.width() );
-    img.scaledToWidth( scaleSize ).save( &buffer, "webp" );
 
-    memcpy( data.data(), ba.data(), ba.size() );
-    data.resize( ba.size() );
```

Comment:
Don't revert anything else except swap the order of those 2 lines.

QBuffer control the content of the `QByteArray ba`. The original code involves an implicit conversion `QByteaArray->char*`.

Also, `&std::vector::front` is a crash if the vector is empty. It generally shouldn't occur in post-c++11 code.